nilearn/glm/first_level/first_level.py

```diff
@@ -1050,6 +1132,23 @@ def first_level_from_bids(
         derivatives_folder=derivatives_folder,
     )
 
+    kwargs_load_confounds = _check_kwargs_load_confounds(**kwargs)
+
+    if drift_model is not None and kwargs_load_confounds is not None:
+        if "high_pass" in kwargs_load_confounds.get("strategy"):
+            if drift_model == "cosine":
+                verb = "duplicate"
+            if drift_model == "polynomial":
+                verb = "conflict with"
+
+            warn(
+                f"""Confounds will contain a high pass filter,
+ that may {verb} the {drift_model} one used in the model.
+ Remember to visualize your design matrix before fitting your model
+ to check that your model is not overspecified.""",
```

Comment:
Not sure about the phrasing of the warning.

Reply:
looks reasonable to me

Reply:
To me toot.